cardano-rt-view/src/Cardano/Benchmarking/RTView/NodeState/Updater.hs

```diff
@@ -195,20 +200,19 @@ itIsErrorMessage aMeta =
 updateNodeUpTime :: NodeState -> UTCTime -> NodeState
 updateNodeUpTime ns now = ns { nsInfo = newNi }
  where
-  newNi =
-    currentNi
-      { niStartTime = startTime
-      , niUpTime    = upTime
-      }
+  newNi = currentNi { niUpTime = upTime }
   currentNi = nsInfo ns
-  currentStartTime@(UTCTime day time) = niStartTime currentNi
-  startTime =
-    if day == (ModifiedJulianDay 0) && time == 0
-      then now -- This is the first LogObject from the node, update startTime.
-      else currentStartTime -- Node already sent something earlier, keep startTime.
+  startTime = niStartTime currentNi
   diffBetweenNowAndStart = diffUTCTime now startTime
   upTime = addUTCTime diffBetweenNowAndStart (UTCTime (ModifiedJulianDay 0) 0)
 
+-- | Node sends its start time just once, with its release, version and commit.
+updateNodeStartTime :: NodeState -> Text -> NodeState
+updateNodeStartTime ns startTime = ns { nsInfo = newNi }
+ where
+  newNi = currentNi { niStartTime = read (T.unpack startTime) :: UTCTime }
```

Comment:
should not need to be translated from a text!
can't you trace the start time as a numeric value? something like nanoseconds?